### PR TITLE
Add caseSensitive parameter on the InlineSyntax constructor

### PR DESCRIPTION
**Pull request by chenzhiguang**
Fix: https://github.com/dart-lang/markdown/issues/367

### CODE REVIEW

File: lib/src/inline_parser.dart [hunk 1]
```diff
@@ -353,8 +353,12 @@ abstract class InlineSyntax {
   ///
   /// If [startCharacter] is passed, it is used as a pre-matching check which
   /// is faster than matching against [pattern].
-  InlineSyntax(String pattern, {int? startCharacter})
-      : pattern = RegExp(pattern, multiLine: true),
+  ///
+  /// If [caseSensitive] is disabled, then case is ignored when matching
+  /// the [pattern].
+  InlineSyntax(String pattern, {int? startCharacter, bool? caseSensitive})
```
Comment (srawlins) [inline lib/src/inline_parser.dart:359]: Let's make this a non-nullable bool, defaulting to true.

**Review by srawlins — CHANGES_REQUESTED** [review]
Can you add a few tests which might indicate how this would be used? Thanks for the help!

**Comment by chenzhiguang** [comment]
> Can you add a few tests which might indicate how this would be used? Thanks for the help!

Sorry, just saw this comment, I will add some tests later soon.

**Review by srawlins — APPROVED** [review]
Thanks!

**Pull request merged**